eva-server/src/main/java/uk/ac/ebi/eva/server/ws/RegionWSServerV2.java

```diff
@@ -137,7 +152,49 @@ public ResponseEntity getVariantsByRegion(@PathVariable("regionId") String regio
 
             resourcesList.add(new Resource<>(variant, Arrays.asList(sourcesLink, annotationsLink)));
         });
-        return new ResponseEntity(new Resources<>(resourcesList), HttpStatus.OK);
+
+        PagedResources pagedResources = new PagedResources<>(resourcesList, pageMetadata);
+
+        if (pageable.getPageNumber() > 0) {
+            pagedResources.add(new Link(linkTo(methodOn(RegionWSServerV2.class).getVariantsByRegion(regionId, species,
```

Comment:
When there is code that is repetitive with just some differences try to extract it to a method with the needed parameters, that way the code will be more readable. In this you could use `createPaginationLink` and use it for creating the first, last, next and previous links

Reply:
Fixed :)